mmv1/third_party/terraform/data_sources/data_source_dns_record_set.go

```diff
@@ -1,86 +1,145 @@
 package google
 
 import (
+	"context"
 	"fmt"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"google.golang.org/api/dns/v1"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 )
 
-func dataSourceDnsRecordSet() *schema.Resource {
-	return &schema.Resource{
-		Read: dataSourceDnsRecordSetRead,
+var _ datasource.DataSource = &GoogleDnsRecordSetDataSource{}
 
-		Schema: map[string]*schema.Schema{
-			"managed_zone": {
-				Type:     schema.TypeString,
-				Required: true,
-			},
+func NewGoogleDnsRecordSetDataSource() datasource.DataSource {
+	return &GoogleDnsRecordSetDataSource{}
+}
 
-			"name": {
-				Type:     schema.TypeString,
-				Required: true,
-			},
+// GoogleDnsRecordSetDataSource defines the data source implementation
+type GoogleDnsRecordSetDataSource struct {
+	client  *dns.Service
+	project types.String
+}
 
-			"rrdatas": {
-				Type:     schema.TypeList,
-				Computed: true,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
-				},
-			},
+type GoogleDnsRecordSetModel struct {
+	Id          types.String `tfsdk:"id"`
+	ManagedZone types.String `tfsdk:"managed_zone"`
+	Name        types.String `tfsdk:"name"`
+	Rrdatas     types.List   `tfsdk:"rrdatas"`
+	Ttl         types.Int64  `tfsdk:"ttl"`
+	Type        types.String `tfsdk:"type"`
+	Project     types.String `tfsdk:"project"`
+}
 
-			"ttl": {
-				Type:     schema.TypeInt,
-				Computed: true,
-			},
+func (d *GoogleDnsRecordSetDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dns_record_set"
```

Comment:
(Moving review comments across from stacked PR)

This is a source of an issue - TypeName needs to be "the full data source type, including the provider type prefix and an underscore. For example, examplecloud_thing." ([from docs](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.1.1/datasource#MetadataResponse)).

Even when using the [google-beta provider the types still only start with `google_`](https://github.com/hashicorp/terraform-provider-google-beta/blob/4a7212618248e0dcd8dcb94cf4a3d9bbd04757bf/google-beta/provider.go#L941) not `google-beta_`

However, the value of `req.ProviderTypeName` is "google-beta" so the provider is sending a `datasource.MetadataResponse` with `google-beta_dns_record_set` as the type name

Bet that's causing an issue somewhere, but I'm not sure where!

Reply:
Addressed in 267067ae70cdf59a0a93e3a2033b248ecad9aafa